paper/sections/hash-and-resubmit.tex

```diff
@@ -191,26 +191,23 @@ \section{The Hash-and-Resubmit Pattern}
 \includegraphics[width=0.9 \columnwidth]{figures/har-example.pdf}
 \end{center}
 \vspace*{-5mm}
-\caption{Gas-cost reduction using the \emph{hash-and-resubmit} pattern (lower
+\caption{Gas-cost reduction of Algorithm~\ref{alg:game-storage} using the \emph{hash-and-resubmit} pattern (lower
 is better). By avoiding gas-heavy storage operations, the aggregated cost of
 \textsf{submit} and \textsf{contest} is decreased by 95\%.}
 \label{fig:har-example}
 \vspace*{-5mm}
 \end{figure}
 
-\noindent \textbf{Consequences.} The most obvious consequence of applying the
-\emph{hash-and-resubmit} pattern is the circumvention of storage
-structures, a benefit that saves a substantial amount of gas, especially when
-these structures are large. To that extent, smart contracts that exceed the
-Ethereum block gas limit and benefit sufficiently for the alleviation of
-storage structures are becoming practical. Furthermore, the pattern enables
-off-chain transactions, significantly improving the performance of smart
-contracts.
+\noindent \textbf{Consequences.} The consequence of applying the
+\emph{hash-and-resubmit} pattern is the circumvention of storage,
+a benefit that saves a substantial amount of gas, especially when
+stored structures are large. Therefore, smart contracts that exceed the
+Ethereum block gas limit and benefit sufficiently from the alleviation of
+storage can become practical.
 
 \noindent \textbf{Known uses.} To our knowledge, we are the first to address
-consistency and availability by combining blockchain's transparency with data
-structures signatures in a manner that eliminates storage variables from
-smart contracts.
+consistency and availability by combining blockchain's transparency with commitments
```

Comment:
Should we now mention summa as a known use?

Reply:
Yes, sure -- although I think this pull request should be merged separately, as it pertains to editorial changes.

Reply:
Yes, right.